pkg/binlog/event/event.go

```diff
@@ -159,6 +159,33 @@ func GenFormatDescriptionEvent(header *replication.EventHeader, latestPos uint32
 	return ev, errors.Trace(err)
 }
 
+// GenRotateEvent generates a RotateEvent.
+// ref: https://dev.mysql.com/doc/internals/en/rotate-event.html
+func GenRotateEvent(header *replication.EventHeader, latestPos uint32, nextLogName []byte, position uint64) (*replication.BinlogEvent, error) {
```

Comment:
what's difference between `latestPos` and `position`?

Reply:
`latestPos` is the previous end position of the latest events or the start position of _this_ event.
`position` is the rotate target position for _this_ event or the start position of the _next_ event.